sites/shared/components/account/sets.mjs

```diff
@@ -110,495 +109,486 @@ export const NewSet = ({
-    evt.preventDefault()
-    if (evt.target.value !== value) {
-      setValue(evt.target.value)
-    }
-  }
-
-  const save = async () => {
-    startLoading()
-    const result = await backend.updateSet(mset.id, { notes: value })
-    if (result.success) {
-      refresh()
-      toast.for.settingsSaved()
-    } else toast.for.backendError()
-    stopLoading()
-  }
-
-  // Shared props for tabs
-  const tabProps = { activeTab, setActiveTab, t }
-
-  return (
-    <div>
-      <div className="tabs w-full">
-        <Tab id="edit" {...tabProps} />
-        <Tab id="preview" {...tabProps} />
-      </div>
-      <div className="flex flex-row items-center mt-4">
-        {activeTab === 'edit' ? (
-          <textarea
-            rows="5"
-            className="textarea textarea-bordered textarea-lg w-full"
-            placeholder={t('placeholder')}
-            onChange={update}
-            value={value}
-          />
-        ) : (
-          <div className="text-left px-4 border w-full">
-            <Markdown>{value}</Markdown>
-          </div>
-        )}
-      </div>
-      <div className="my-2 flex gap-2 items-center justify-center">
-        <button className="btn btn-secondary" onClick={save} disabled={value === mset.name}>
-          <span className="flex flex-row items-center gap-2">
-            {loading ? (
-              <>
-                <Spinner />
-                <span>{t('processing')}</span>
-              </>
-            ) : (
-              t('save')
-            )}
-          </span>
-        </button>
-      </div>
-    </div>
-  )
-}
+export const Mset = ({ id, publicOnly = false }) => {
+  // Hooks
+  const { account, control } = useAccount()
+  const { setLoadingStatus, LoadingStatus } = useLoadingStatus()
+  const backend = useBackend()
+  const { t } = useTranslation(ns)
+  const router = useRouter()
+  const { locale } = router
+  const docs = useSetDocs(locale)
+  const measieDocs = useMeasurementDocs(locale)
 
-const EditUnits = ({ t, mset, backend, toast, refresh, startLoading, stopLoading }) => {
-  const [selection, setSelection] = useState(mset?.imperial === true ? 'imperial' : 'metric')
+  // Context
+  const { setModal } = useContext(ModalContext)
 
-  const update = async (val) => {
-    setSelection(val)
-    const asBool = val === 'imperial' ? true : false
-    if (asBool !== mset.imperial) {
-      startLoading()
-      const result = await backend.updateSet(mset.id, { imperial: asBool })
+  const [filter, setFilter] = useState(false)
+  const [edit, setEdit] = useState(false)
+  const [mset, setMset] = useState()
+  // Set fields for editing
+  const [name, setName] = useState(mset?.name)
+  const [image, setImage] = useState(mset?.image)
+  const [isPublic, setIsPublic] = useState(mset?.public ? true : false)
+  const [imperial, setImperial] = useState(mset?.imperial ? true : false)
+  const [notes, setNotes] = useState(mset?.notes || '')
+  const [measies, setMeasies] = useState({})
```

Comment:
personally would prefer to save the original set in the state as, like, `origSet`, and a deep copy to manipulate in a single state or maybe even ref object `mset`. then just to write an onChange callback that sets each of these properties on the `mset` so you don't have to store and mess with them all separately. Then in the `save` callback, you loop through all the properties to compare `origSet` to `mset` to send the changes

Reply:
no side-quests